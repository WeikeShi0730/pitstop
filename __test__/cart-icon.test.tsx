import {
  render,
  RenderResult,
  screen,
  fireEvent,
} from "@testing-library/react";
import CartIcon from "../components/cart-icon.component";

let documentBody: RenderResult;
describe("<CartIcon />", () => {
  it("shows empty in <CartIcon />", () => {
    documentBody = render(<CartIcon cartItems={[]} />);
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });

  it("shows content in <CartIcon />", () => {
    documentBody = render(
      <CartIcon
        cartItems={[
          {
            product: {
              imageUrl: "test",
              name: "",
              price: 0,
              id: "",

              featured: false,
              sold: 0,
              teamId: "",
            },
            count: 10,
          },
          {
            product: {
              imageUrl: "test",
              name: "",
              price: 0,
              id: "",

              featured: false,
              sold: 0,
              teamId: "",
            },
            count: 5,
          },
        ]}
      />
    );
    expect(documentBody.queryByText("15")).toBeInTheDocument();
  });

  it("shows <CartIcon /> on click", () => {
    documentBody = render(<CartIcon cartItems={[]} />);
    expect(screen.queryByText("Checkout")).not.toBeInTheDocument();
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(screen.queryByText("Checkout")).toBeInTheDocument();
  });
});
