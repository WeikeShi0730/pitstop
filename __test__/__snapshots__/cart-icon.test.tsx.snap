// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartIcon /> shows empty in <CartIcon /> 1`] = `
<body>
  <div>
    <div
      class="relative text-sm md:text-base"
    >
      <button
        class="flex items-center justify-center space-x-2 font-light hover:underline-primary hover:underline-offset-4"
      >
        <svg
          fill="none"
          height="1em"
          id="cartIcon"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="9"
            cy="21"
            r="1"
          />
          <circle
            cx="20"
            cy="21"
            r="1"
          />
          <path
            d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
          />
        </svg>
        <div
          class="px-1 w-8 flex justify-start"
        >
          <div />
        </div>
      </button>
    </div>
  </div>
</body>
`;
